### PR TITLE
Qvdang/vz2300

### PR DESCRIPTION
**Pull request by qvdang**
Replacing with a more secure hash algorithm.

### CODE REVIEW

File: pkg/vmo/secret.go [hunk 1]
```diff
@@ -50,9 +51,11 @@ func (hp HashedPasswords) Bytes() (passwordBytes []byte) {
 
 // SetPassword set a password for a user with a hashing algo
 func (hp HashedPasswords) SetPassword(name, password string) (err error) {
-	prefix := "{SHA}"
-	hash := hashSha(password)
-	hp[name] = prefix + hash
+	hashValue, ok := hashBcrypt(password)
+	if ok != nil {
```
Comment (pfmackin) [inline pkg/vmo/secret.go:55]: can you use err, not ok

Comment (qvdang) [reply pkg/vmo/secret.go]: Done.

File: pkg/vmo/secret.go [hunk 2]
```diff
@@ -32,11 +31,13 @@ const (
 // HashedPasswords name => hash
 type HashedPasswords map[string]string
 
-func hashSha(password string) string {
-	s := sha1.New()
-	s.Write([]byte(password))
-	passwordSum := s.Sum(nil)
-	return base64.StdEncoding.EncodeToString(passwordSum)
+func hashBcrypt(password string) (string, error) {
+	hashedPassword, err := bcrypt.GenerateFromPassword([]byte(password), bcrypt.DefaultCost)
```
Comment (wmhopkins) [inline pkg/vmo/secret.go:35]: I assume `bcrypt.DefaultCost` is the number of rounds for the bcrypt algorithm? Did you research how many rounds is considered "secure", and compare that with the default value? What did you find?

**Review by wmhopkins — COMMENTED** [review]
Looks good to me, but would like to understand the `DefaultCost` value relative to secure best practice. Do the GPS secure coding guidelines say anything about that?

File: pkg/vmo/secret.go (hunk 2 above)
Comment (qvdang) [reply pkg/vmo/secret.go:35]: Yes, DefaultCost is the cost factor(CF) currently set to 10 in the library.  It is the password expansion rounds(2^n).  Obviously, the higher the cost factor, the more secure the hash is.  However unless you know exactly what kind of hardware your env is running, it's not easy to choose a cost factor that balances out security and acceptable performance.  That was the reason I went with the default.  I did try various CF such as 14,15,16 but the responses were significantly slower on my OCI Linux host env.
GPS does not mention about the CF of the Bcrypt algorithm.  Let me know if you have any suggestion on the CF that we should use other than the default.

Comment (wmhopkins) [reply pkg/vmo/secret.go:35]: Seems reasonable to me.

**Review by wmhopkins — APPROVED** [review]
Ship it!

**Pull request merged**